### DISCUSSION TITLE
Asset Register

### DISCUSSION THREAD

**Discussion by zyinhls** [stac-utils/titiler-pgstac#78] [General]
Hi Vincent or any experts,

If there are many assets in database, I just want to register three assets. How to put three assets in the below register command?

curl -X 'POST' 'http://127.0.0.1:8081/mosaic/register' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -d '{"collections":["landsat-c2l2-sr"], "bbox":[-123.75,34.30714385628804,-118.125,38.82259097617712], "filter-lang": "cql-json"}' | jq

Thanks.

**Comment by vincentsarago** [comment]
I can't remember if we can filter the assets directly with the CQL filter 
Maybe @bitner knows :-)

**Reply by zyinhls** [reply]
Thanks, Vincent. 
I use eoAPI-raster-Swagger UI register format with assets B04, B03, B02 (below). It registers without error, but it cannot filter assets and all 12 assets still available. What does assets in metadata mean? What is the difference between bbox and bounds inside metadata?

curl -X 'POST' 'http://0.0.0.0:8082/mosaic/register' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -d '{
  "collections":["HLSS30.v2.0"], 
  "bbox":[-180,-90,180,90], 
  "filter-lang": "cql-json", 
  "metadata": {
                "type": "mosaic",
                "bounds": [
                 -180,
                 -90,
                 180,
                 90
                ],
                "minzoom": 8,
                "maxzoom": 12,
                "name": "HLSS30.v2.0",
                "assets": [
                 "B04",
                 "B03",
                 "B02"
                ],
                "defaults": {}
              }
  }' | jq

**Reply by vincentsarago** [reply]
I see! The `assets` you added is in the metadata part which is not taken into account by default see https://github.com/stac-utils/titiler-pgstac/issues/64

> What does assets in metadata mean?

The `metadata` are informative and not used in the endpoints. Please see https://stac-utils.github.io/titiler-pgstac/advanced/metadata/ for more information 

> What is the difference between bbox and bounds inside metadata?

bbox is part of the pgstac filter, bounds is part of the metadata

I'm also not quite sure what you are trying to do.
- do you want the tile endpoint to default to a combination of the 3 assets (B04,B03,B02)?
- do you want users to only be able to create combination between those 3 assets (adding asset filters in the CQL query)

**Reply by zyinhls** [reply]
Thank you for metadata information. For example, in register, we can use bbox and datetime to filter data. Can we filter assets when register?  If there are 10 assets: a1, a2, a3, a4... a10.  Can we register to filter to 3 assets like a2, a3, a4 (not use other assets)? If that, maybe it will be faster.
